tracer/src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -702,6 +709,16 @@ public bool DiagnosticSourceEnabled
         /// </summary>
         internal ImmutableAzureAppServiceSettings? AzureAppServiceMetadata { get; set; }
 
+        /// <summary>
+        /// Gets or sets a value indicating whether to calculate the peer.service tag from predefined precursor attributes when using the v0 schema.
+        /// </summary>
+        internal bool PeerServiceTagsEnabled { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether to remove the service names when using the v0 schema.
+        /// </summary>
+        internal bool RemoveClientServiceNamesEnabled { get; set; }
+
```

Comment:
nit: in a [config telemetry PR](https://github.com/DataDog/dd-trace-dotnet/pull/4219), I've been working on making our `internal` properties in `TracerSettings` be `get`-only where possible, as otherwise we have to track their "initial" values, and record the changes subsequently 😩. It makes testing a bit more of a pain, but reduces allocation/work that would otherwise be required... 

Looks like it's easy to change here, but it will break existing tests, which would need to set them indirectly via config instead

```suggestion
        /// <summary>
        /// Gets a value indicating whether to calculate the peer.service tag from predefined precursor attributes when using the v0 schema.
        /// </summary>
        internal bool PeerServiceTagsEnabled { get; }

        /// <summary>
        /// Gets a value indicating whether to remove the service names when using the v0 schema.
        /// </summary>
        internal bool RemoveClientServiceNamesEnabled { get; }

```

Reply:
Updated in 22fc5b495cb6bbf627e84246e4b433f9fc84b260